### DISCUSSION TITLE
Caching Pages with CDN and NextJS SSG

### DISCUSSION THREAD

**Discussion by KobeTwo** [vercel/next.js#39648] [Help]
Hi,
we are using NextJS with an Azure Front Door CDN in front. 
Because we have world wide user base we already cache complete pages (not only static files) in the CDN which are generated by NextJS with SSG and invalidated with ISR.
We already faced problems with displaying an old page after a deployment (because the page is cached in CDN) and requesting old js resources which are not cached/invalidated in the CDN and after the deployment not available in the server anymore. This issue was relatively easy to solve by pushing all static resource versions to central storage.
Now we are facing the related issue that the old page cached by the CDN cannot retrieve the data via /_next/data/${buildid} because it is not available any more after deployment.
Do you have good recommendations on how to solve this?
Considerations we already took:
- also keep a history of the deployments for a certain amount of time -> seems like too much overhead (costs, complex routing) 
- cache only the static content and never the pages (with our worldwide user base we are afraid of the latencies and performance impact)

**Comment by garnerp** [comment]
Purge the Azure Front Door CDN cache for every deployment?

**Reply by KobeTwo** [reply]
Would also be an option. But we deploy really frequently and this would take out a lot of benefits the CDN currently offers with caching things long term.
